### PR TITLE
ST-295: Added landing page for update estimated income journey

### PR DESCRIPTION
**Pull request by BriWak**
- [x] Unit tests passing
- [x] Coverage reached
- [x] Acceptance tests passing
- [x] Reviewed

### CODE REVIEW

File: test/views/html/incomes/EstimatedPayLandingPageSpec.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  val empId = 1
+  val employerName = "fakeFieldValue"
+
+  "Estimated Pay Landing Page" should {
+    behave like pageWithBackLink
+    behave like pageWithCancelLink(Call("GET",controllers.routes.IncomeSourceSummaryController.onPageLoad(empId).url))
+    behave like pageWithCombinedHeader(
+      messages("tai.incomes.edit.preHeading", employerName),
+      messages("tai.incomes.edit.preHeading", employerName))
```
Comment (SeanArmstrong) [inline test/views/html/incomes/EstimatedPayLandingPageSpec.scala]: This doesnt look right?

File: app/views/main.scala.html [hunk 2]
```diff
@@ -173,7 +173,7 @@
     Map[String, Any](
         "pageTitle" -> s"${title} ${serviceSuffix}",
         "linkElems" -> Map(
-            "url" -> controllers.routes.AssetsController.versioned("stylesheets/tai03092018.min.css")
```
Comment (SeanArmstrong) [inline app/views/main.scala.html]: This is slightly odd but probably not worth worrying about for now.

I assume that the base branch is out of date with master and therefore this branch also is.

File: app/views/incomes/estimatedPayLandingPage.scala.html [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    <div>
+        <p>@messages("tai.incomes.landing.intro")</p>
+        <h2>@messages("tai.incomes.landing.subheading", employerName)</h2>
+    </div>
+    <p>
+        @Html(messages(
+                "tai.incomes.landing.you.can",
+                Link.toInternalPage(
+                    url=controllers.employments.routes.EndEmploymentController.employmentUpdateRemove(empId).toString,
+                    value=Some(messages("tai.incomes.landing.employment.ended.link", employerName)),
```
Comment (SeanArmstrong) [inline app/views/incomes/estimatedPayLandingPage.scala.html]: From slack
```
Play i18n messages do not work well with HTML, unfortunately.  You would have to surround the `@messages(...)` with `@Html(...)`, but that then means the employer name is vulnerable to XSS (potentially).  So it's not ideal at all.
```

Comment (BriWak) [reply app/views/incomes/estimatedPayLandingPage.scala.html]: As discussed, this has been used elsewhere in TAI and presumably the Pen testing would have picked it up if it was an issue. Is it ok to leave this as is?

File: test/views/html/incomes/EstimatedPayLandingPageSpec.scala [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      doc(view).getElementsByTag("p").text must include(messages("tai.incomes.landing.intro"))
+      doc(view) must haveLinkWithText(messages("tai.incomes.landing.employment.ended.link", employerName))
+      doc must haveLinkWithUrlWithID("updateEmployer",
+        controllers.employments.routes.EndEmploymentController.employmentUpdateRemove(empId).url)
+      doc(view).getElementsByClass("button").text must include(messages("tai.income.details.updateTaxableIncome.update"))
+    }
+  }
+
+
+  override def view: Html = views.html.incomes.estimatedPayLandingPage(employerName, empId, isPension = false)
```
Comment (SeanArmstrong) [inline test/views/html/incomes/EstimatedPayLandingPageSpec.scala]: Need to check for when it is a pension, sorry :(

Comment (BriWak) [reply test/views/html/incomes/EstimatedPayLandingPageSpec.scala]: Done

**Pull request merged**